File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int z = 0,ans =0,l = 0;
        for(int i = 0; i < nums.size() ; i++){
            z += (nums[i] == 0);
            while(z > 1)
                z -= nums[l++] == 0;
            ans = max(ans,i - l);
        } 
        return ans;
    }
};

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.rb
# @param {Integer[]} nums
# @return {Integer}
def longest_subarray(nums)
    z, l, ans = 0, 0, 0 
    nums.each_with_index do |num, i|
        z += 1 if num == 0
        while z > 1
            z -= 1 if nums[l] == 0
            l += 1
        end
        ans = [ans, i - l].max
    end
    ans 
end
